gliderpy/plotting.py

```diff
@@ -90,3 +90,55 @@ def plot_transect(
     ax.set_ylim(ax.get_ylim()[0], 0)
 
     return fig, ax
+
+
+@register_dataframe_method
+def plot_ctd(
+    df: pd.DataFrame,
+    profile_number: int,
+    var: str,
+    ax: plt.Axes = None,
+    color: str | None = None,
+) -> tuple:
+    """Make a CTD profile plot of pressure vs property
+    depending on what variable was chosen.
+
+    :param profile_number: profile number of CTD
+    :param var: variable to plot against pressure
+    :param ax: existing axis to plot on (default: None)
+    :param color: color for the plot line (default: None)
+    :return: figure, axes
+    """
+    g = df.groupby(["longitude", "latitude"])
+    profile = g.get_group(list(g.groups)[profile_number])
+
+    if ax is None:
+        fig, ax = plt.subplots(figsize=(5, 6))
+        ax.plot(profile[var], profile["pressure"], label=var, color=color)
+        ax.set_ylabel("Pressure")
+        ax.set_xlabel(var)
+        ax.invert_yaxis()
```

Comment:
This should be an if-check block outside of the ax creation. We had it before if I'm not mistaken.

Reply:
The problem occurs when calling the legend while using the twiny axis. We have three scenarios:

**1. When using just one axis (not referring to a second one like ax2), the legend will only display the last variable plotted if twiny is used, so we need to create twoo box one for each legend and specify where they should be placed.**

`
@register_dataframe_method
def plot_ctd(
    df: pd.DataFrame,
    profile_number: int,
    var: str,
    ax: plt.Axes = None,
    color: str | None = None,
) -> tuple:
    """Make a CTD profile plot of pressure vs property
    depending on what variable was chosen.

    :param profile_number: profile number of CTD
    :param var: variable to plot against pressure
    :param ax: existing axis to plot on (default: None)
    :param color: color for the plot line (default: None)
    :return: figure, axes
    """
    g = df.groupby(["longitude", "latitude"])
    profile = g.get_group(list(g.groups)[profile_number])

    if ax is None:
        fig, ax = plt.subplots(figsize=(5, 6))
        ax.plot(profile[var], profile["pressure"], label=var, color=color)
        ax.set_ylabel("Pressure")
        ax.set_xlabel(var)
        ax.invert_yaxis()
        ax.legend(loc="lower center", bbox_to_anchor=(0.5, 0.07))
        return fig, ax
    else:
        ax = ax.twiny()  # Create a new x-axis on top
        ax.plot(profile[var], profile["pressure"], label=var, color=color)
        ax.set_xlabel(var)
        ax.legend(loc="lower center")
        return ax.figure, ax
fig, ax = plot_ctd(df, 0, var="temperature", color="blue")
fig, ax = plot_ctd(df, 0, var="salinity", ax=ax, color="red")`



**2. Or calling the legend inside each clouse (if ax is none and else) without defining each ax they would overlap eachother.**


**3. Or we defined each ax and call the legens like this so they can be in the same box**

`@register_dataframe_method
def plot_ctd(
    df: pd.DataFrame,
    profile_number: int,
    var: str,
    ax: plt.Axes = None,
    color: str | None = None,
) -> tuple:
    """Make a CTD profile plot of pressure vs property
    depending on what variable was chosen.

    :param profile_number: profile number of CTD
    :param var: variable to plot against pressure
    :param ax: existing axis to plot on (default: None)
    :param color: color for the plot line (default: None)
    :return: figure, axes
    """
    g = df.groupby(["longitude", "latitude"])
    profile = g.get_group(list(g.groups)[profile_number])

    if ax is None:
        fig, ax = plt.subplots(figsize=(5, 6))
        lns1 = ax.plot(profile[var], profile["pressure"], label=var, color=color)
        ax.set_ylabel("Pressure")
        ax.set_xlabel(var)
        ax.invert_yaxis()
        
        # Get handles and labels for the legend
        handles, labels = ax.get_legend_handles_labels()
        ax.legend(handles, labels, loc="best")
        return fig, ax
    else:
        ax2 = ax.twiny()  # Create a new x-axis on top
        lns2 = ax2.plot(profile[var], profile["pressure"], label=var, color=color)
        ax2.set_xlabel(var)
        
        # Get handles and labels for both axes
        handles1, labels1 = ax.get_legend_handles_labels()
        handles2, labels2 = ax2.get_legend_handles_labels()
        
        # Combine handles and labels
        handles = handles1 + handles2
        labels = labels1 + labels2
        
        # Set the combined legend
        ax.legend(handles, labels, loc="best")
        
        return ax.figure, ax2

# Example usage:
fig, ax = plot_ctd(df, 0, var="temperature", color="blue")
fig, ax = plot_ctd(df, 0, var="salinity", ax=ax, color="red")
`